### PR TITLE
Fix Go tool version to 1.22

### LINKED ISSUE #15524: Zbctl (latest) is not working with SaaS
**Describe the bug**

The latest zbctl version (from 8.3.0 upwards) is not working with SaaS, at least we are not able to fetch a token from INT stage. Reported by @sdorokhova [here](https://camunda.slack.com/archives/CSQ2E3BT4/p1702023006959649) 

Be aware that older versions work.

**To Reproduce**

> [!Note]
> Make sure to remove the .camunda/credentials file or move it, before testing this.

* download latest zbctl
* Create a cluster
* create a client in SaaS
* copy the env vars
* run any zbctl command
* see that it times out

**Expected behavior**

We can connect with zbctl to SaaS.

**Log/Stacktrace**

```
Error: rpc error: code = Canceled desc = failed to apply token: failed to obtain access token: Post "https://login.cloud.ultrawombat.com/oauth/token": context deadline exceeded
```

</p>
</details>

**Environment:**
- Zeebe Version: 8.3.0+
- Configuration: SaaS

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

This hardwires the Go version to use for building `zbctl` to '1.22'.
This resolves a network issue for `zbctl` on Windows that was fixed in Go '1.22.0'.

## Related issues

closes #15524 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by tmetzke** [comment]
Should we backport this to 8.4 and 8.3? We haven't backported https://github.com/camunda/zeebe/issues/15958 (bump Go from `1.19` to `1.21`).

**Comment by nicpuppa** [comment]
> Should we backport this to 8.4 and 8.3? We haven't backported #15958 (bump Go from `1.19` to `1.21`).

Usually bump version are not backported. As zbctl is backword compatible, in my opinion is not necessary. Customer can use the new version of zbctl. But maybe someone else have a different opinion 👀

**Review by nicpuppa — APPROVED** [review]
LGTM

❓ The bump of the linter is mandatory with the new go version ?

**Comment by tmetzke** [comment]
> > Should we backport this to 8.4 and 8.3? We haven't backported #15958 (bump Go from `1.19` to `1.21`).
> 
> Usually bump version are not backported. As zbctl is backword compatible, in my opinion is not necessary. Customer can use the new version of zbctl. But maybe someone else have a different opinion 👀

We have a regression in 8.3 and 8.4 though for zbctl on Windows. But it works if you extend the timeout. It's horribly slow though without the version bump.

> LGTM
> 
> ❓ The bump of the linter is mandatory with the new go version ?

It doesn't seem to be mandatory regarding our test suite at least. According to the release notes, support for Go version `1.22` comes with linter version `1.56.0`, whatever means in detail.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -384,7 +384,7 @@ jobs:
         uses: golangci/golangci-lint-action@v4
         with:
           # fixed to avoid triggering false positive; see https://github.com/golangci/golangci-lint-action/issues/535
-          version: v1.55.2
+          version: v1.56.2
```
Comment (tmetzke) [inline .github/workflows/ci.yml:387]:
```suggestion
          version: v1.55.2
```

**Comment by tmetzke** [comment]
The artifacts created by this change in a release dry-run work fine for me.
I added [a comment](https://github.com/camunda/zeebe/issues/15524#issuecomment-1953897751) to the issue and attached test artifacts.

**Comment by tmetzke** [comment]
Changes based on the decisions documented in [the issue](https://github.com/camunda/zeebe/issues/15524#issuecomment-1953819704).

**Review by nicpuppa — APPROVED** [review]
LGTM 🚀

**Comment by tmetzke** [comment]
Manually backported this to `stable/8.4` with #16490

**Pull request merged**